### ISSUE TITLE
Docker run issue

### ISSUE THREAD

**Issue by shijbian** [QingZHANG98/INFO550_project#2]
- [x] The repository's README explained how to download the Docker image.
- [x] The README explained how to build and retrieve the report.
- [x] The report built without error and could be retrieved.

**Comment by QingZHANG98** [comment]
You need a dash before "it", which means the interactive mode. Because I had an entry point of 'make report', it's no need to use this option.
Just copy the code in Readme.

**Comment by shijbian** [comment]
> You need a dash before "it", which means the interactive mode. Because I had an entry point of 'make report', it's no need to use this option.
> Just copy the code in Readme.

Hi Qing,

Thank you so much! Everything looks great now. I have updated the comment.

Best,
Shijia

**Comment by shijbian** [comment]
Hi Qing, 

just two more comments, but let me know if there is something I did not do properly.

1. To generate the html report interactively, I used the bash command in terminal as being shown below. Tag "old_pandoc" should be added to the command line: `docker run -it qingzh11/info550:old_pandoc /bin/bash`
2. I wanted to extract the html output at my local ~/Desktop. I ran the command line below:
`docker run -v ~/Desktop:/INFO550_project/report/report_Rmd qingzh11/info550:old_pandoc`
However, nothing was happening. I guess this was because you did not include `CMD make -f Makefile` in your Dockerfile. However, let me know if you could successfully ran this command line and retrieve your report at your local from your laptop.

**Comment by QingZHANG98** [comment]
1. If you have run `docker build -t info550 .`, it will create a new image info550:latest, and you can call info550 in the following steps. I did not add CMD into info550:old_pandoc, otherwise, it would always automatically run CMD and stop the container. 
2. Maybe you should not use the relative path.  I had the same error when I used "~/my_proj_dir" but was successful with the absolute path. That's why I used `docker run -v ${PWD}:/INFO550_project info550` .` ${PWD} `will return the current path, for me it's '/Users/qing/INFO550_project' in Mac. I have tested in Windows(WSL2)/Linux/macOS, it worked.

**Comment by shijbian** [comment]
1. In this homework, we are not supposed to git clone your repo, docker build the docker image and retrieve the report. What we just need to do was docker pull your docker image from docker hub. The docker image you pushed to docker hub was the one tagged with old_pandoc. Therefore, when I retrieved the report using the docker image I pulled from your docker hub, I had to add the tag, which is `docker run -it qingzh11/info550:old_pandoc /bin/bash`. This worked perfectly fun on my side.
2. It still does not work with absolute path. I am not sure if this is because I am using Mac with Apple M1 chip. My question is how the report could be automatically retrieved without `CMD make -f Makefile` in the Dockerfile. 

<img width="902" alt="Screen Shot 2021-12-10 at 1 54 20 PM" src="https://user-images.githubusercontent.com/6429151/145626743-c5f573ad-6ab6-46ea-af3f-90f155efd867.png">

**Comment by QingZHANG98** [comment]
I'm sorry about my confusing reply.  For some reason, It's necessary to clone the repo in github and build a new image. The data in the image will be overwritten by the current repo.
I used renv to synchronize the packages into container and commit it to my image old_pandoc, then we can make report in the container. Sadly, I found docker based on windows will not activate the environment to run CMD.  So I deleted CMD in old_pandoc and put it into the dockerfile of building info550. This dockerfile in github has `CMD make report` . The report will be **built and retrieved based on the local repo** instead of the files in image.
You don't have to pull the old_pandoc.  In docker build it will do
- pull old_pandoc (all packages installed)  +
- read the local codes and data +
- activate environment (I used restore in case I want to use more packages. But all are installed now) +
- make report and overwrite the local file
All steps are put in "Execute the analysis". I think it's more flexible but really strange.
